### PR TITLE
UIKit Refactor: Bower to NPM Migration, Partial JS Refactor, and New JS Build Pipeline

### PR DESCRIPTION
**Pull request by sghoweri**
Here’s the big one I’ve been waiting a long, long time to finally get into PL core 🎉

There’s a bunch of small, under the hood changes here (see commits for the specifics) but in a nutshell:

1. Migrate Pattern Lab’s UIKit [away from Bower](https://github.com/pattern-lab/styleguidekit-assets-default/issues/76) and fully onto NPM
2. Reorganize UIKit’s JavaScript into 3 main buckets to help with separating out the different concerns: the main JS entrypoints (which didn’t previously exist as physical files, based on how things had been set up), components, and utils.
3. Painstakingly go through every single JS file in PL and make sure every file properly pulls in and manages their own dependencies required
4. Fix a TON of broken JS code — undefined or improperly scoped variables and the like
5. Adds a very minimal Webpack build to bundle and minify the JS.

Lots more to come based off of this work but really just trying to start small to things keep things in working order + get us in a good spot to start working on some much more abitious work on the UI side of things!

Would love any help I can get to further test this out to make sure all the kinks and gotchas have been ironed out!

CC @pattern-lab/devs

### CODE REVIEW

**Comment by sghoweri** [comment]
By the way @bmuenzenmeyer, before you bring it up, I’d really love to run all these JS files through Prettier — I just wanted to wait till the overall code changes here were understood and OK’d first so we aren’t looking at 10x the number of JS changes 😉

**Comment by bmuenzenmeyer** [comment]
Love it.  Will provide a more comprehensive review ASAP.

**Comment by bmuenzenmeyer** [comment]
@sghoweri this is knock-down drag-out awesome work. Thank you so much for putting so much time and energy into it!


Am I testing this correctly:

from your branch...

```
$ npm install && npm run bootstrap
$ cd packages/uikit-workshop
$ npm run build

> @pattern-lab/uikit-workshop@1.0.0-alpha.7 build C:\src\patternlab-node-pr\packages\uikit-workshop
> gulp && npx webpack-cli --config webpack.config.js --progress

[10:27:52] Using gulpfile C:\src\patternlab-node-pr\packages\uikit-workshop\gulpfile.js[10:27:52] Starting 'build:css'...
[10:27:53] Starting 'clean:html'...
[10:27:53] Finished 'clean:html' after 46 ms
[10:27:53] Starting 'build:html'...
[10:27:53] Finished 'build:html' after 250 ms
Autoprefixer's process() method is deprecated and will removed in next major release. Use postcss([autoprefixer]).process() instead
[10:27:54] Finished 'build:css' after 1.99 s
[10:27:54] Starting 'default'...
[10:27:54] Finished 'default' after 58 μs
npx: installed 111 in 20.417s
Cannot find module 'webpack'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @pattern-lab/uikit-workshop@1.0.0-alpha.7 build: `gulp && npx webpack-cli --config webpack.config.js --progress`
npm ERR! Exit status 1
```

**Comment by sghoweri** [comment]
Hmm 🤔 that sounds about right.

I can take on look on my end once I’m back from lunch but off the top of my head I’m gonna guess that we might need to add webpack-cli as a dependency to our package.json file. 

Maybe I had that globally installed?

**Comment by sghoweri** [comment]
@bmuenzenmeyer - my latest push should fix that error you mentioned! Sorry about that -- totally missed that I hadn't wired that up to use a locally installed version of `webpack-cli` vs a globally installed version.

That part should now be good to go -- I'm now taking a quick peek at the merge conflicts on this.

**Comment by sghoweri** [comment]
@bmuenzenmeyer OK - I've gone in and resolved the couple of merge conflicts to get this up to date with the latest changes on dev, plus adjusted a couple small things in the JS based on some additional testing I've done on this.

There's a few CSS-related UI issues I'm aware of that I'll try to tackle this weekend (not yet sure if they're related to this PR itself or if they just existing issues -- if they're new it's almost certainly just a CleanCSS config to turn things down a little) but I'm pretty good about the JavaScript changes here all in all + how much better we'll be off once we start tackling some even more ambitious UI work 😉

**Comment by bmuenzenmeyer** [comment]
@sghoweri this is building now and looks good on a quick glance on the 🚌 
going to give it a closer look this tonight / weekend though.

nice work man! 💃 🕺 🌮 🍺 🎺 🥁 🎸

**Comment by bmuenzenmeyer** [comment]
@sghoweri I've been travelling so this is taking a bit longer. Appreciate the patience!

**Comment by sghoweri** [comment]
@bmuenzenmeyer no worries - appreciate the update!

File: packages/uikit-workshop/package.json [hunk 1]
```diff
@@ -18,6 +17,11 @@
       "name": "Brian Muenzenmeyer",
       "homepage": "http://brianmuenzenmeyer.com",
       "role": "Developer"
+    },
+    {
+      "name": "Salem Ghoweri",
```
Comment (bmuenzenmeyer) [inline packages/uikit-workshop/package.json]: Well deserved!

File: packages/uikit-workshop/views/partials/general-footer.mustache [hunk 2]
```diff
@@ -46,11 +46,12 @@
 </script>
 
 <script id="pl-js-insert-{{ cacheBuster }}">
+  /*! loadJS: load a JS file asynchronously. [c]2014 @scottjehl, Filament Group, Inc. (Based on http://goo.gl/REQGQ by Paul Irish). Licensed MIT */
```
Comment (bmuenzenmeyer) [inline packages/uikit-workshop/views/partials/general-footer.mustache]: ❤️

**Pull request merged**